student/views.py

```diff
@@ -1,3 +1,28 @@
-# from django.shortcuts import render
+from django.shortcuts import render, HttpResponseRedirect
+from .models import Student
+from .forms import UpdateStudentAccountSettingsForm
 
-# Create your views here.
+
+def account_update_success(request):
+    context = {}
+    add_is_student_to_context(request, context)
+    return render(request, 'account_settings_update_success.html', context)
+
+
+def update_student_account_settings_view(request):
+    context = {}
+    try:
+        student_object = Student.objects.get(user_id=request.user.id)
+    except Student.DoesNotExist:
+        return render(request, "dead_end.html")
+    update_student_form = UpdateStudentAccountSettingsForm(request.POST or None, instance=student_object)
+    if update_student_form.is_valid():
+        update_student_form.save()
+        return HttpResponseRedirect("/account_update_success/")
+    context["form"] = update_student_form
+    add_is_student_to_context(request, context)
+    return render(request, "student_account_settings.html", context)
```

Comment:
Try to be consistent and either user `render()` and `redirect()` or `TemplateResponse` and `HttpResponseRedirect`.